File: java/FizzBuzzWithMaven/src/main/java/de/welcz/RuleBuilder.java
package de.welcz;

/**
 * Fluent interface builder for {@link Rule}
 */
class RuleBuilder {
    private int denominator;
    private String result;

    RuleBuilder forDenominator(int denominator) {
        this.denominator = denominator;
        return this;
    }

    RuleBuilder givingResult(String result) {
        this.result = result;
        return this;
    }

    Rule create() {
        return new Rule(denominator, result);
    }
}

File: typescript/fizzbuzz-mocha/src/rule_builder.ts
import {Rule} from "./rule";

export class RuleBuilder {
    private denominators: number[] = [];

    public forDenominator(denominator: number): RuleBuilder {
        this.denominators.push(denominator);
        return this;
    }

    public returning(result: string): Rule {
        return new Rule(result, this.denominators);
    }
}
